scalecube-configuration/src/main/java/io/scalecube/configuration/tokens/VaultPathBuilder.java

```diff
@@ -0,0 +1,40 @@
+package io.scalecube.configuration.tokens;
+
+import com.bettercloud.vault.EnvironmentLoader;
+import io.scalecube.configuration.AppSettings;
+import java.util.Objects;
+
+final class VaultPathBuilder {
+
+  private String vaultSecretPath;
+
+  String getPath(String alias) {
+    return getVaultSecretPath().concat(alias);
+  }
+
+  private String getVaultSecretPath() {
+    if (vaultSecretPath == null) {
+      vaultSecretPath = String.format(
+          getVaultSecretPathTokensKeyPattern(), getVaultKeyValueEngine());
+    }
+    return vaultSecretPath;
+  }
+
+  private String getVaultSecretPathTokensKeyPattern() {
+    String vaultSecretPathKeyPattern = AppSettings
+        .builder()
+        .build()
+        .getProperty("vault.secret.path");

```

Comment:
I'm not sure that we should reread system and file properties every time.
Maybe will be better use it like in this PR (https://github.com/scalecube/scalecube-organization-service/pull/19/files#diff-6502b431d8e0018fe0307cc2b06acb6a)